libs/platform/src/lib/components/link/link.component.ts

```diff
@@ -0,0 +1,124 @@
+import {
+    AfterViewInit,
+    Component,
+    ChangeDetectionStrategy,
+    EventEmitter,
+    ElementRef,
+    Input,
+    OnInit,
+    Output,
+    ViewChild
+} from '@angular/core';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+
+export type LinkType = 'standard' | 'emphasized';
+export type NavigationTarget = '_blank' | '_self' | '_parent' | '_top' | 'framename';
+const VALID_INPUT_TYPES = ['standard', 'emphasized'];
+
+@Component({
+    selector: 'fdp-link',
+    templateUrl: './link.component.html',
+    styleUrls: ['./link.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LinkComponent implements OnInit, AfterViewInit {
+    emphasized: boolean = false;
+    isfocused: boolean = false;
+
+    /** Access child element, for checking link content*/
+    @ViewChild('link', { read: ElementRef, static: false })
+    anchor: ElementRef;
+
+    /** Id for the link */
+    @Input()
+    id?: string;
+
+    /** href value to Navigate to */
+    @Input()
+    href: string;
+
+    /** target where navigation will happen, Default=same frame */
+    @Input()
+    target?: NavigationTarget;
+
+    /** type of link, options standard or Emphasized, Default=standard */
+    @Input()
+    type: LinkType = 'standard';
+
+    @Input()
+    get disabled(): boolean {
+        return this._disabled;
+    }
+
+    /** Link enable or disabled status */
+    set disabled(value: boolean) {
+        this._disabled = coerceBooleanProperty(value);
+    }
+
+    @Input()
+    get inverted(): boolean {
+        return this._inverted;
+    }
+
+    /** set incase of Inverted link */
+    set inverted(value: boolean) {
+        this._inverted = coerceBooleanProperty(value);
+    }
+
+    /** Tooltip text to show when focused for more than  timeout value*/
+    @Input()
+    title?: string;
+
+    /** Specifies the language of the linked document */
+    @Input()
+    hreflang?: string;
+
+    /** Specifies that the target will be downloaded when a user clicks on the hyperlink */
+    @Input()
+    download?: string;
+
+    /** Specifies what media/device the linked document is optimized for */
+    @Input()
+    media?: string;
+
+    /** Specifies the relationship between the current document and the linked document */
+    @Input()
+    rel?: string;
+
+    /** Emitting link click event */
+    @Output()
+    click: EventEmitter<any> = new EventEmitter();
```

Comment:
Should have type

Reply:
this event is possible on MouseEvent as well as KeyboardEvent. keeping as "any" so both cases can be handled.

Reply:
@DeepakSap14 this can be done using `event: MouseEvent | KeyboardEvent`

Reply:
Sure. I will be using TouchEvent as well with MouseEvent | KeyboardEvent, as this is also expected.